music-player/src/components/AddSongToPlaylist.tsx

```diff
@@ -0,0 +1,80 @@
+import { Modal, Button } from 'antd';
+import * as React from 'react';
+import { connect } from 'react-redux';
+import { Dispatch } from 'redux';
+
+import { State } from '../types';
+import { getPlaylists } from '../reducers';
+import { getAddSongToPlaylistAction } from '../actions';
+
+class AddSongToPlaylist extends React.Component<
+  {
+    playlistsNotIncludeSongIndex: string[];
+    realIndex: number;
+    dispatch: Dispatch;
+  },
+  { show: boolean }
+> {
+  constructor(props: any) {
+    super(props);
+    this.state = { show: false };
+  }
+
+  closeWindowFunction = () => this.setState({ show: false });
+
+  render = () => {
+    const { playlistsNotIncludeSongIndex, dispatch } = this.props;
+    return (
+      <span>
+        <Button
+          className="song-button add-song-button"
+          shape="circle"
+          icon="plus"
+          size="small"
+          onClick={() => this.setState({ show: true })}
+        />
+        <Modal
+          title="Add Song to Playlist"
+          visible={this.state.show}
+          onCancel={this.closeWindowFunction}
+          footer={[
+            <Button key="ok" type="primary" onClick={this.closeWindowFunction}>
+              OK
+            </Button>,
+          ]}
+        >
+          {!this.state.show
+            ? ''
+            : playlistsNotIncludeSongIndex.length === 0
+              ? 'All playlists have the song.'
+              : playlistsNotIncludeSongIndex.map((playlist: string) => (
+                  <div key={playlist}>
+                    <Button
```

Comment:
It's weird UI to use multiple buttons.
Why not checkboxes?

Reply:
I was thinking about that. Usually when someone adds song to playlist he would want to add it only to one playlist.
So I gave up on checkboxes, I thought maybe radio buttons but it means more clicking and it looks a bit old.
Anyway, I can change it to checkboxes.